### PR TITLE
feat: added tpl to enable templated config values

### PR DESCRIPTION
**Pull request by kvanzuijlen**
We use [helm secrets](https://github.com/jkroepke/helm-secrets) to encrypt configuration values like credentials. To reference these it would be nice to be able to reference these in the configuration without creating a custom configmap. To allow for this the tpl function has been added in this PR.

### CODE REVIEW

**Review by viceice — COMMENTED** [review]
What are the side effects? Breaking change?

**Comment by kvanzuijlen** [comment]
Hi @viceice,

It could be breaking, but I think it's _higly_ unlikely. It would be breaking in the case that someone out there has the following  in their renovate configuration:
```
"someKey": "{{anyValue}}"
```
or
```
"someKey": "{{anyValue"
```

If you can think of a valid configuration key that would likely have such a value my suggestion would be to change it to a conditional option (add a key to the values.yaml called `configEnableTpl` or something like that).

For configurations not containing `{{` the config will still work the exact same, although it might be unnoticeably slower.

Please let me know what your preferred solution would be, either one should work for us and anyone having the same requirements as us in the future.

**Comment by viceice** [comment]
renovate template files can contain handlebar token

**Comment by viceice** [comment]
https://docs.renovatebot.com/templates/

**Comment by kvanzuijlen** [comment]
Ofcourse... I didn't think about that, sorry. I think it's either back to the drawing board or making it clear in the documentation that with `configEnableHelmTpl` set to true you have to escape renovate templates like:
`"{{"{{depName}}"}}"` (wich would result in `"{{depName}}"` in the config.json).

Using `configEnableHelmTpl` would make this a useful feature to include imo, but it would add an extra section to the Helm chart documentation. I think the benefit will outweigh this drawback, since this feature would probably only be used consciously.

Can you give me a go ahead to implement `configEnableHelmTpl`, or would you like to see another solution (I can't think of one from the top of my head)?

**Review by viceice — CHANGES_REQUESTED** [review]
go ahead

File: charts/renovate/values.yaml [hunk 1]
```diff
@@ -57,6 +57,13 @@ renovate:
   #     "repositories": ["username/repo", "orgname/repo"]
   #   }
 
+  # -- Use the Helm tpl function on your configuration
+  # Allows you to reference values using "{{ .Values.someValue }}" in your config
+  # NOTE: using this means that you have to escape your config entries in the config
+  # key above containing {{ in the value by wrapping it like this:
+  # "key": "{{ "{{depName}}" }}"
```
Comment (viceice) [inline charts/renovate/values.yaml:64]: Can you please add a new section with this content to the readme, so it's more visible?

Comment (kvanzuijlen) [reply charts/renovate/values.yaml:64]: Should I keep it here as well?

Comment (viceice) [reply charts/renovate/values.yaml:64]: No, i think readme is enough

File: charts/renovate/README.md.gotmpl [hunk 2]
```diff
@@ -47,3 +47,10 @@ The slim suffix will be added to the tag if not present. To disable this behavio
 ## Redis
 
 Please checkout [bitnami redis](https://artifacthub.io/packages/helm/bitnami/redis) chart for additional redis configuration.
+
+## renovate.configEnableHelmTpl
```
Comment (viceice) [inline charts/renovate/README.md.gotmpl:51]:
```suggestion
## Renovate config templating

Enable `renovate.configEnableHelmTpl` to use helm templates for generated renovate `config.json`.
```
what about this ? seems a better heading and description

Comment (kvanzuijlen) [reply charts/renovate/README.md.gotmpl:51]: Sure!

File: charts/renovate/README.md [hunk 3]
```diff
@@ -102,3 +103,11 @@ The slim suffix will be added to the tag if not present. To disable this behavio
 ## Redis
 
 Please checkout [bitnami redis](https://artifacthub.io/packages/helm/bitnami/redis) chart for additional redis configuration.
+
+## renovate.configEnableHelmTpl
```
Comment (viceice) [inline charts/renovate/README.md:107]: needs an update too. 😉

**Review by viceice — APPROVED** [review]

**Comment by viceice** [comment]
will be published with next renovate release

**Pull request merged**